src/components/health-questionnaire/index.tsx

```diff
@@ -26,21 +34,35 @@ const HealthQuestionnaire = (): JSX.Element => {
   };
 
   const handleSubmit = (): void => {
-    router.push(ROUTES.appointment_caregiver_list);
+    onNext();
   };
 
   return (
-    <Background>
-      <QuestionnaireContainer>
-        {currentStep === STEPS.STEP_1 && <Step1 onNext={handleNext} stepKey={STEPS.STEP_1} />}
-        {currentStep === STEPS.STEP_2 && (
-          <Step2 onNext={handleNext} onBack={handleBack} stepKey={STEPS.STEP_2} />
-        )}
-        {currentStep === STEPS.STEP_3 && (
-          <Step3 onNext={handleSubmit} onBack={handleBack} stepKey={STEPS.STEP_3} />
-        )}
-      </QuestionnaireContainer>
-    </Background>
+    <>
+      {diagnoses && activities && capabilities ? (
+        <QuestionnaireContainer>
+          {currentStep === STEPS.STEP_1 && (
+            <Step1 onNext={handleNext} stepKey={STEPS.STEP_1} diagnoses={diagnoses} />
+          )}
+          {currentStep === STEPS.STEP_2 && (
+            <Step2
+              onNext={handleNext}
+              onBack={handleBack}
+              stepKey={STEPS.STEP_2}
+              activities={activities}
+            />
+          )}
+          {currentStep === STEPS.STEP_3 && (
+            <Step3
+              onNext={handleSubmit}
+              onBack={handleBack}
+              stepKey={STEPS.STEP_3}
+              capabilities={capabilities}
+            />
+          )}
+        </QuestionnaireContainer>
+      ) : null}
+    </>
```

Comment:
I don't think it's a problem, i use it here so that user didn't see questionnaire without questions, other way it will render empty, then rerender with questions when request finished and user will see element blinking

Reply:
about fragment in this case its fine. with null, I'm not sure, have you tried `??` or `&&` instead of `?:`